src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.InProcess.cs

```diff
@@ -177,14 +183,16 @@ internal abstract partial class AbstractNavigateToSearchService
         private static void AddResultIfMatch(
             Document document, DeclaredSymbolInfo declaredSymbolInfo,
             PatternMatcher nameMatcher, PatternMatcher containerMatcherOpt,
+            ISet<string> kinds,
             ArrayBuilder<PatternMatch> nameMatches, ArrayBuilder<PatternMatch> containerMatches, 
             ArrayBuilder<SearchResult> result, CancellationToken cancellationToken)
         {
             nameMatches.Clear();
             containerMatches.Clear();
 
             cancellationToken.ThrowIfCancellationRequested();
-            if (nameMatcher.AddMatches(declaredSymbolInfo.Name, nameMatches) &&
+            if (kinds.Contains(GetItemKind(declaredSymbolInfo)) &&
```

Comment:
i would love if we could avoid the check if we knew htat Kinds matched the full list of Kinds we support.  Consider potentially having this only be non-null if it actually would ever reduce the set.  That way we don't incur hte lookup on *every* string name in the world.  

Nav to is super compute-costly.  So avoiding unnecessary stuff would be really desirable.

Reply:
Effectively, if the kinds passed in, intersected with the kinds we support, equals the kinds we support, then we just don't do any filtering.  We would only filter if hte kinds requested was a subset of what we support.

Reply:
The enumset test should be as cheap as creating a fast path for cases where filtering isn't necessary, so I'll probably focus on that approach.

Reply:
➡️ Fixed in 0371b07edc8ba5b804b211ce34fa4ca043726b7a